src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfilesDebugLaunchProvider.cs

```diff
@@ -48,9 +51,22 @@ internal class LaunchProfilesDebugLaunchProvider : DebugLaunchProviderBase, IDep
         /// <summary>
         /// Called by CPS to determine whether we can launch
         /// </summary>
-        public override Task<bool> CanLaunchAsync(DebugLaunchOptions launchOptions)
+        public override async Task<bool> CanLaunchAsync(DebugLaunchOptions launchOptions)
         {
-            return TplExtensions.TrueTask;
+            try
+            {
+                // Launch providers to enforce requirements for debuggable projects
+                await QueryDebugTargetsInternalAsync(launchOptions, true);
+            }
+            catch(Exception e)
+            {
+                if (string.Compare(e.Message, VSResources.ProjectNotRunnableDirectly, StringComparison.OrdinalIgnoreCase) == 0)
+                {
+                    return await TplExtensions.FalseTask;
```

Comment:
>return await TplExtensions.FalseTask; [](start = 20, length = 37)

You should be able to simply return false; (or return true;) in this method

Reply:
I agree. Will be fixed in the next commit.